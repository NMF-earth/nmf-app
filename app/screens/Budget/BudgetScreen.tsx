import React from "react";
import { ScrollView } from "react-native";
import { useSelector } from "react-redux";
import styles from "./BudgetScreen.styles";
import { Text, Button } from "../../components";
import { NumberOfDaysVegetarian, ProgressChart } from "./components";
import { t } from "../../utils";
import { navigate } from "../../navigation";
import { selectors } from "./ducks";

interface Props {
  navigation: {
    push: (screen: string) => void;
  };
}

const BudgetScreen = (props: Props) => {
  const navigator = navigate(props.navigation);
  const monthlyCarbonBudget = useSelector(selectors.getMonthlyCarbonBudget);
  const totalCurrentMonthEmissions = useSelector(
    selectors.getCurrentMonthAllCarbonValue
  );
  const transportCurrentMonthEmissions = useSelector(
    selectors.getCurrentMonthTransportCarbonValue
  );
  const foodCurrentMonthEmissions = useSelector(
    selectors.getCurrentMonthFoodCarbonValue
  );
  const otherCurrentMonthEmissions = useSelector(
    selectors.getCurrentMonthCustomCarbonValue
  );

  const totalCurrentYearEmissions = useSelector(
    selectors.getCurrentYearAllCarbonValue
  );
  const transportCurrentYearEmissions = useSelector(
    selectors.getCurrentMonthTransportCarbonValue
  );
  const foodCurrentYearEmissions = useSelector(
    selectors.getCurrentYearFoodCarbonValue
  );
  const otherCurrentYearEmissions = useSelector(
    selectors.getCurrentYearCustomCarbonValue
  );

  return (
    <ScrollView style={styles.container}>
      <ProgressChart
        isMonth
        totalEmissions={totalCurrentMonthEmissions}
        foodEmissions={foodCurrentMonthEmissions}
        transportEmissions={transportCurrentMonthEmissions}
        otherEmissions={otherCurrentMonthEmissions}
        monthlyEmissionsBudget={monthlyCarbonBudget}
      />
      <Button.Primary
        style={styles.monthlyBudgetButton}
        fullWidth
        textType={"Primary"}
        onPress={navigator.openMontlyBudget}
      >
        <Text.Primary bold center white>
          {t("BUDGET_SCREEN_SET_MONTHLY_BUDGET")}
        </Text.Primary>
      </Button.Primary>
      <ProgressChart
        totalEmissions={totalCurrentYearEmissions}
        foodEmissions={foodCurrentYearEmissions}
        transportEmissions={transportCurrentYearEmissions}
        otherEmissions={otherCurrentYearEmissions}
        monthlyEmissionsBudget={monthlyCarbonBudget}
      />
      <NumberOfDaysVegetarian />
    </ScrollView>
  );
};

export default BudgetScreen;
