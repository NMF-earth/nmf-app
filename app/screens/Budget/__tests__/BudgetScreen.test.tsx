import React from "react";
import { create } from "react-test-renderer";

import BudgetScreen from "../BudgetScreen";

const props = {
  navigation: {
    push: () => {
      // do nothing.
    },
  },
};

it("BudgetScreen renders correctly", () => {
  const tree = create(<BudgetScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// TODO: add test to verify that monthly budget is opened
