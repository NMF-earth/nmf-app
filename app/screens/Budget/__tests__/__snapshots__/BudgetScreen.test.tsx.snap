// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <ProgressChart
      foodEmissions={[MockFunction]}
      isMonth={true}
      monthlyEmissionsBudget={[MockFunction]}
      otherEmissions={[MockFunction]}
      totalEmissions={[MockFunction]}
      transportEmissions={[MockFunction]}
    />
    <Button.Primary
      fullWidth={true}
      onPress={[Function]}
      style={
        Object {
          "marginVertical": 6,
        }
      }
      textType="Primary"
    >
      <Text.Primary
        bold={true}
        center={true}
        white={true}
      >
        Set monthly budget
      </Text.Primary>
    </Button.Primary>
    <ProgressChart
      foodEmissions={[MockFunction]}
      monthlyEmissionsBudget={[MockFunction]}
      otherEmissions={[MockFunction]}
      totalEmissions={[MockFunction]}
      transportEmissions={[MockFunction]}
    />
    <NumberOfDaysVegetarian />
  </View>
</RCTScrollView>
`;
